tensorflow/python/keras/layers/local_test.py

```diff
@@ -45,84 +48,84 @@ def test_locallyconnected_1d(self):
         for strides in [1]:
           if padding == 'same' and strides != 1:
             continue
-          for data_format in ['channels_first', 'channels_last']:
-            for implementation in [1, 2]:
-              kwargs = {
-                  'filters': filters,
-                  'kernel_size': filter_length,
-                  'padding': padding,
-                  'strides': strides,
-                  'data_format': data_format,
-                  'implementation': implementation
-              }
-
-              if padding == 'same' and implementation == 1:
-                self.assertRaises(ValueError,
-                                  keras.layers.LocallyConnected1D,
-                                  **kwargs)
-              else:
-                testing_utils.layer_test(
-                    keras.layers.LocallyConnected1D,
-                    kwargs=kwargs,
-                    input_shape=(num_samples, num_steps, input_dim))
-
-  def test_locallyconnected_1d_regularization(self):
+          for implementation in [1, 2]:
+            kwargs = {
+                'filters': filters,
+                'kernel_size': filter_length,
+                'padding': padding,
+                'strides': strides,
+                'data_format': data_format,
+                'implementation': implementation
+            }
+
+            if padding == 'same' and implementation == 1:
+              self.assertRaises(ValueError,
+                                keras.layers.LocallyConnected1D,
+                                **kwargs)
+            else:
+              testing_utils.layer_test(
+                  keras.layers.LocallyConnected1D,
+                  kwargs=kwargs,
+                  input_shape=(num_samples, num_steps, input_dim))
+
+  @parameterized.parameters(['channels_first', 'channels_last'])
+  def test_locallyconnected_1d_regularization(self, data_format):
     num_samples = 2
     num_steps = 8
     input_dim = 5
     filter_length = 3
     filters = 4
-    for data_format in ['channels_first', 'channels_last']:
-      for padding in ['valid', 'same']:
-        for implementation in [1, 2]:
+    for padding in ['valid', 'same']:
+      for implementation in [1, 2]:
+        kwargs = {
+            'filters': filters,
+            'kernel_size': filter_length,
+            'kernel_regularizer': 'l2',
+            'bias_regularizer': 'l2',
+            'activity_regularizer': 'l2',
+            'data_format': data_format,
+            'implementation': implementation,
+            'padding': padding
+        }
+
+        if padding == 'same' and implementation == 1:
+          self.assertRaises(ValueError,
+                            keras.layers.LocallyConnected1D,
+                            **kwargs)
+        else:
+          with self.cached_session():
```

Comment:
Rather than using `self.cached_session`, please use the `run_all_keras_modes` decorator

Reply:
run_all_keras_modes cause  absl.testing.parameterized.Duplicate Test NameError, so will use self.cached_session.